docker-java-api/src/main/java/com/github/dockerjava/api/model/GenericResource.java

```diff
@@ -1,36 +1,56 @@
 package com.github.dockerjava.api.model;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
 import lombok.EqualsAndHashCode;
 import lombok.ToString;
 
+import javax.annotation.CheckForNull;
 import java.io.Serializable;
 
+/**
+ * @since {@link RemoteApiVersion#VERSION_1_24}
+ */
+@JsonTypeInfo(
+    use = JsonTypeInfo.Id.NAME,
+    include = JsonTypeInfo.As.WRAPPER_OBJECT
+)
+@JsonSubTypes({@JsonSubTypes.Type(
+    value = NamedResourceSpec.class,
+    name = "NamedResourceSpec"
+), @JsonSubTypes.Type(
+    value = DiscreteResourceSpec.class,
+    name = "DiscreteResourceSpec"
+)})
 @EqualsAndHashCode
 @ToString
 public abstract class GenericResource<T> extends DockerObject implements Serializable {
     private static final long serialVersionUID = 1L;
 
+    /**
+     * @since 1.24
+     */
     @JsonProperty("Kind")
     String kind;
-    @JsonProperty("Value")
-    T value = null;
 
+    /**
+     * @see #kind
+     */
+    @CheckForNull
     public String getKind() {
         return kind;
     }
 
-    public GenericResource withKind(String kind) {
+    /**
+     * @see #kind
+     */
+    public GenericResource<T> withKind(String kind) {
         this.kind = kind;
         return this;
     }
 
-    public T getValue() {
-        return value;
-    }
+    public abstract T getValue();
```

Comment:
Note to any potential reviewers: This is causing the build to fail because it is an API change; the method used to be concrete but is now abstract. But IMO it's a valid change. It isn't going to impact any users because no one was constructing these `GenericResource`s before anyway. There was no way to do so.

I suppose I could leave it non-abstract, keep a `T value` property here, and make the child classes basically empty shells with declarations only. But it really doesn't make any sense for users to be able to construct `GenericResource` instances, this _should_ be abstract, if you tried to pass one to the docker API it would fail.

All of that to say, I don't know how to bypass the API change check so that the build can succeed.